media-video/wireplumber/wireplumber-9999.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( lua5-{3,4} )
+
+inherit lua-single meson
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://gitlab.freedesktop.org/pipewire/${PN}.git"
+	EGIT_BRANCH="master"
+	inherit git-r3
+else
+	SRC_URI="https://gitlab.freedesktop.org/pipewire/${PN}/-/archive/${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="Replacement for pipewire-media-session"
+HOMEPAGE="https://gitlab.freedesktop.org/pipewire/wireplumber"
+
+LICENSE="MIT"
+SLOT="0/0.4"
+IUSE="systemd"
+
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+
+# introspection? ( dev-libs/gobject-introspection ) is valid but likely only used for doc building
+BDEPEND="
+	dev-libs/glib
+	dev-util/gdbus-codegen
+	dev-util/glib-utils
+"
+
+DEPEND="
+	${LUA_DEPS}
+	>=dev-libs/glib-2.62
+	>=media-video/pipewire-0.3.26
+	virtual/libc
+	systemd? ( sys-apps/systemd )
+
+"
+
+# Any dev-lua/* deps get declared like this inside RDEPEND:
+#	$(lua_gen_cond_dep '
+#		dev-lua/<NAME>[${LUA_USEDEP}]
+#	')
+RDEPEND="${DEPEND}"
+
+DOCS=( {NEWS,README}.rst )
+
+src_configure() {
+	local emesonargs=(
+		-Dintrospection=disabled # Likely only used for doc generation
+		-Dsystem-lua=true # We always unbundle everything we can
+		#-Dsystem-lua-version=<version number> # Some distros can't enforce deterministc system Lua themselves
+		$(meson_feature systemd)
+		-Dsystemd-system-service=false # Matches upstream
+		$(meson_use systemd systemd-user-service)
+		#-Dsystemd-system-unit-dir=""
+		#-Dsystemd-user-unit-dir=""
+	)
+
+	meson_src_configure
+}
+
+src_install() {
+	# No idea if this is needed with meson but eclass example has this for emake
+	LUA_VERSION="$(lua_get_version)" meson_src_install
```

Comment:
You can test by installing both (5.3 and 5.4 which you said is preferred upstream) and then see if it uses 5.3 when you express a preference for it.

But LUA_VERSION is probably just an example for some random Makefile. Likely not necessary here.